📡 Add support for WalletConnect

thesan opened the pull request:
Partially address:
- #4729

thesan commented:
This PR is already way too big so I changed the target to a new branch `wallet-connect` branch so this can be merged before this whole feature is stable, i.e wallet connect support and better mobile support.

kdembler reviewed (commented):
I will review the code tomorrow. From using, all looks great. Only issue I've noticed is the same as in Atlas PR - disconnecting a wallet in Pioneer doesn't end WC session. If I select WC option again, my previous connection is restored and I don't have an option to connect different wallet through WC

thesan commented:
@kdembler 
> disconnecting a wallet in Pioneer doesn't end WC session.

I just fixed this.

FYI there was a change on this PR to show the member selection modal with the "Disconnect wallet" button when you click on "Join Now". However it was bundled with a lot of UI changes so I moved everything to #4781 (to have a more manageable PR size). So until we get to #4781, if you connect with a wallet with no memberships you can disconnect the wallet by clicking on "Add New Proposal".

kdembler commented on packages/ui/src/accounts/model/walletConnect.ts:
```diff
@@ -0,0 +1,189 @@
+import { Signer } from '@polkadot/api/types'
+import { WalletConnectModal } from '@walletconnect/modal'
+import { SessionTypes } from '@walletconnect/types'
+import { UniversalProvider } from '@walletconnect/universal-provider'
+import type { UniversalProvider as Provider } from '@walletconnect/universal-provider/dist/types/UniversalProvider.d.ts'
```
Aren't those 2 imports the same?

kdembler commented on packages/ui/src/accounts/model/walletConnect.ts:
```diff
@@ -0,0 +1,189 @@
+  }
+
+  public getAccounts = async (): Promise<WalletAccount[]> => {
+    return Promise.resolve(this._accounts ?? [])
```
Nitpick, but you don't need the `Promise.resolve`, same below

kdembler commented on packages/ui/src/app/components/ExtensionWarning.tsx:
```diff
@@ -1,6 +1,6 @@
 import React, { useEffect, useState } from 'react'
 
-import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
+// import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
```
Can be removed?
